ls8/server/api/messages/messages.controller.js

```diff
@@ -0,0 +1,125 @@
+const MessageModel = require("../models/message.model");
+const { Filtering } = require("../utils/filterMessages");
+const uniqid = require('uniqid');
+
+
+
+exports.get_messages_handler = async (req, res, next) => {
+  const params = req.query;
+  const dbData = await MessageModel.find({deletedAt: undefined})
```

Comment:
```suggestion
  const dbData = await MessageModel
      .find({deletedAt: undefined})
      .sort([req.query.sortBy]: req.query.sortValue)
      .skip(req.query.skip)
      .limit(req.query.limit)
      .lean()
      .exec();
```

Из базы надо доставать меньше данных и уже с нужными критериями, нежели доставать всю базу в память и там уже что-то делать. С твоим подходом сервер не будет держать нагрузку.